### PR TITLE
Use formik task7

### PR DESCRIPTION
**Pull request by MimimiBB-8**
- Used formik for validation field in form
- Used request to backend

### CODE REVIEW

File: src/components/BasicForm/BasicForm.tsx [hunk 1]
```diff
@@ -1,53 +1,235 @@
...
+  { label: 'Thriller', value: 'thriller' },
+  { label: 'Biography', value: 'biography' },
+  { label: 'Action', value: 'action' },
+  { label: 'Sci-Fi', value: 'sci-fi' },
 ]
 
-function BasicForm() {
-  const [value1, setValue1] = useState<SelectOption[]>([options[0]])
+interface BasicFormProps {
+  edidForm?: boolean
```
Comment (tlgadrkn) [inline src/components/BasicForm/BasicForm.tsx:24]: editForm

File: src/components/BasicForm/BasicForm.tsx [hunk 2]
```diff
@@ -1,53 +1,235 @@
...
+  onClick: () => void
+  showModalWindow?: any
+}
+
+interface LLL {
```
Comment (tlgadrkn) [inline src/components/BasicForm/BasicForm.tsx:29]: please put meaningful name

File: src/components/BasicForm/BasicForm.tsx [hunk 3]
```diff
@@ -1,53 +1,235 @@
...
+  id?: number
+  title: string,
+  // eslint-disable-next-line camelcase
```
Comment (tlgadrkn) [inline src/components/BasicForm/BasicForm.tsx:32]: you can configure eslint to get rid of these comments

File: src/components/BasicForm/BasicForm.tsx [hunk 4]
```diff
@@ -1,53 +1,235 @@
...
+      .required('Rating is required'),
+    runtime: Yup.number().
+      integer()
+      .positive('Runtime must be greater than zero')
+      .required('Runtime is required'),
+  })
+
+  let movieDescription: string | any[] = []
+
+  if (edidForm !== false) {
```
Comment (tlgadrkn) [inline src/components/BasicForm/BasicForm.tsx:70]:
```suggestion
  if (editForm) {
```

File: src/components/BasicForm/BasicForm.tsx [hunk 5]
```diff
@@ -1,53 +1,235 @@
...
+  })
+
+
+
+  const addSelectOption = (data: any) => {
+    const elem: any[] = [];
+    selectValue.map((item: any) => {
+      elem.push(item.label)
+    })
+    data.genres = elem
```
Comment (tlgadrkn) [inline src/components/BasicForm/BasicForm.tsx:120]: you must use dispatch action for setting state in redux

File: src/components/BasicForm/FormItem.tsx [hunk 6]
```diff
@@ -1,19 +1,31 @@
...
+      <label className={style.input_title}>
+        {title}
+        <input
+          className={style.input_item}
+          id={id}
+          type={type}
+          placeholder={placeholder}
+          defaultValue={value}
+          onChange={onchange}
+          name={name}
```
Comment (tlgadrkn) [inline src/components/BasicForm/FormItem.tsx:25]: you forgot to pass `value` prop

**Review by tlgadrkn — APPROVED** [review]
form validations don't work.

File: src/components/BasicForm/BasicForm.tsx (hunk 1 above)
Comment (MimimiBB-8) [reply src/components/BasicForm/BasicForm.tsx]: fixed

File: src/components/BasicForm/BasicForm.tsx (hunk 2 above)
Comment (MimimiBB-8) [reply src/components/BasicForm/BasicForm.tsx]: fixed

**Pull request merged**